scenario_gym/metrics/trajectory.py

```diff
@@ -64,3 +68,1126 @@ def _step(self, state: State) -> None:
+                    entity_class = entity.__class__.__name__
+                    entity_delta_v = (
+                        self.predefined_masses[ego_class]
+                        / (
+                            self.predefined_masses[ego_class]
+                            + self.predefined_masses[entity_class]
+                        )
+                    ) * (np.linalg.norm(vel_ego) - np.linalg.norm(vel_non_ego))
+
+                    # state.t inclusion in the dictionary (line below). To
+                    # track/timestamp TTC in simulation.
+                    if state.t not in self.delta_v_dict:
+                        self.delta_v_dict[state.t] = []
+                    self.delta_v_dict[state.t].append(
+                        {"delta_v": {f"{entity}": entity_delta_v}}
+                    )
+                else:
+                    continue
+
+    def get_state(self):
+        """Return delta_v dictionary."""
+        return self.delta_v_dict
+
+
+# TODOASAP - CONFLICT INDEX
+class ConflictIndex(Metric):
+    """
+    A metric to estimate the conflict index between two agents.
+
+    Incorporates both collision probability and severity factors.
+
+    The conflict index (CI) is calculated based on
+    the predicted change in kinetic energy (ΔKe)
+    in a hypothetical collision scenario between two agents, A1 and A2,
+    at the time they enter and exit a designated conflict area.
+
+    The formula for CI is:
+
+        CI(A1, A2, CA, α, β) = (α * ΔKe) / (e^(β * PET(A1, A2, CA)))
+
+    where:
+        - PET(A1, A2, CA): The Post Encroachment Time between the agents
+        within the conflict area, representing the time
+        until potential collision.
+        - α ∈ [0, 1]: A calibration factor representing the
+        proportion of energy transfer from vehicle body to passengers,
+        used to quantify collision severity.
+        - β (s⁻¹): A calibration factor dependent on scenario factors
+        (e.g., country, road geometry, visibility) that adjusts
+        collision probability estimation.
+        - ΔKe: The absolute change in kinetic energy of the agents
+        before and after the predicted collision,
+        based on masses, velocities, and angles.
+
+    This metric provides an estimation of the collision
+    likelihood and severity in a given conflict scenario.
+    """
+
+    name = "conflict_index"
+
+    def _reset(self, state: State) -> None:
+        self.ego = state.scenario.ego
+        self.t = 0.0
+
+    def calculate(self) -> NotImplementedError:
+        """Calculate the conflict index."""
+        raise NotImplementedError(
+            "The conflict_index metric hasn't been implemented yet. Still a WIP."
+        )
+
+    def _step(self, state: State) -> None:
+        self.calculate
+
+    def get_state(self) -> None:
+        """Return conflict index."""
+        return None
+
+
+# ---------- Collision Check ----------#
+
+
+class CollisionCheck(Metric):
```

Comment:
I guess you want this specifically for the timestamps? If so maybe there could be a better name - I first through it was just a normal collision which we had already, but I see its actually different